src/main/kotlin/com/petqua/domain/oauth/OauthClientProvider.kt

```diff
@@ -0,0 +1,13 @@
+package com.petqua.domain.oauth
+
+import org.springframework.stereotype.Component
+
+@Component
+class OauthClientProvider(oauthClients: Set<OauthClient>) {
+
+    private val clientMapper = oauthClients.associateBy(OauthClient::oauthServerType)
+
+    fun getOauthClient(oauthServerType: OauthServerType): OauthClient {
+        return clientMapper[oauthServerType]!! // TODO 커스텀 예외 생성
+    }
```

Comment:
Map없이 find를 사용하면 어떨까요? null 처리도 용이해 보이네요
```suggestion
class OauthClientProvider(
    private val oauthClients: Set<OauthClient>
) {

    fun getOauthClient(oauthServerType: OauthServerType): OauthClient {
        oauthClients.find { it.oauthServerType() == oauthServerType }
            ?: throw IllegalArgumentException() // TODO 커스텀 예외
    }
}
```